Sources/TemplateKit/AST/TemplateConstant.swift

```diff
@@ -0,0 +1,23 @@
+/// Supported template constants.
+public enum TemplateConstant {
+    case bool(Bool)
+    case int(Int)
+    case double(Double)
+    /// Strings support nested stynax.
+    case string([TemplateSyntax])
```

Comment:
Can this be elaborated a bit more? String is a `String`,  right?

Reply:
in leaf, you can do stuff like this:

```
#("Hello, #(name)")
```